File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> map = new HashMap<>();
        for (String[] cloth : clothes) {
            map.put(cloth[1], map.getOrDefault(cloth[1], 0) + 1);
        }
        for (int cnt : map.values()) {
            answer *= (cnt + 1);
        }
        
        return answer - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let answer = 1;
    const map = new Map();
    
    for (const cloth of clothes) {
        const type = cloth[1];
        map.set(type, (map.get(type) || 0) + 1);
    }
    
    for (const cnt of map.values()) {
        answer *= (cnt + 1);
    }
    
    return answer - 1;
}